stubs/D3DShot/d3dshot/dll/dxgi.pyi

```diff
@@ -0,0 +1,180 @@
+    ) -> _HRESULT: ...
+    def GetFrameDirtyRects(self) -> _HRESULT: ...
+    def GetFrameMoveRects(self) -> _HRESULT: ...
+    def GetFramePointerShape(self) -> _HRESULT: ...
+    def MapDesktopSurface(self) -> _HRESULT: ...
+    def UnMapDesktopSurface(self) -> _HRESULT: ...
+    def ReleaseFrame(self) -> _HRESULT: ...
+
+class IDXGIOutput(IDXGIObject):
+    def GetDesc(self, pDesc: _Pointer[DXGI_OUTPUT_DESC]) -> _HRESULT: ...
+    def GetDisplayModeList(self) -> _HRESULT: ...
+    def FindClosestMatchingMode(self) -> _HRESULT: ...
+    def WaitForVBlank(self) -> _HRESULT: ...
+    def TakeOwnership(self) -> _HRESULT: ...
+    def ReleaseOwnership(self) -> None: ...
+    def GetGammaControlCapabilities(self) -> _HRESULT: ...
+    def SetGammaControl(self) -> _HRESULT: ...
+    def GetGammaControl(self) -> _HRESULT: ...
+    def SetDisplaySurface(self) -> _HRESULT: ...
+    def GetDisplaySurfaceData(self) -> _HRESULT: ...
+    def GetFrameStatistics(self) -> _HRESULT: ...
+
+class IDXGIOutput1(IDXGIOutput):
+    def GetDisplayModeList1(self) -> _HRESULT: ...
+    def FindClosestMatchingMode1(self) -> _HRESULT: ...
+    def GetDisplaySurfaceData1(self) -> _HRESULT: ...
+    def DuplicateOutput(
+        self, pDevice: _Pointer[ID3D11Device], ppOutputDuplication: _Pointer[_Pointer[IDXGIOutputDuplication]]
+    ) -> _HRESULT: ...
+
+class IDXGIAdapter(IDXGIObject):
+    def EnumOutputs(self, Output: UINT, ppOutput: _Pointer[_Pointer[IDXGIOutput]]) -> _HRESULT: ...
+    def GetDesc(self) -> _HRESULT: ...
+    def CheckInterfaceSupport(self) -> _HRESULT: ...
+
+class IDXGIAdapter1(IDXGIAdapter):
+    def GetDesc1(self, pDesc: _Pointer[DXGI_ADAPTER_DESC1]) -> _HRESULT: ...
+
+class IDXGIFactory(IDXGIObject):
+    def EnumAdapters(self) -> _HRESULT: ...
+    def MakeWindowAssociation(self) -> _HRESULT: ...
+    def GetWindowAssociation(self) -> _HRESULT: ...
+    def CreateSwapChain(self) -> _HRESULT: ...
+    def CreateSoftwareAdapter(self) -> _HRESULT: ...
+
+class IDXGIFactory1(IDXGIFactory):
+    def EnumAdapters1(self, Adapter: c_uint, ppAdapter: _Pointer[_Pointer[IDXGIAdapter1]]) -> _HRESULT: ...
+    def IsCurrent(self) -> BOOL: ...
+
+def initialize_dxgi_factory() -> _Pointer[IDXGIFactory1]: ...
+def discover_dxgi_adapters(dxgi_factory: IDXGIFactory1) -> list[_Pointer[IDXGIAdapter1]]: ...
+def describe_dxgi_adapter(dxgi_adapter: IDXGIAdapter1) -> Array[WCHAR]: ...
+def discover_dxgi_outputs(dxgi_adapter: IDXGIAdapter) -> list[_Pointer[IDXGIOutput1]]: ...
+def describe_dxgi_output(dxgi_output: IDXGIOutput) -> _DXGIOutput: ...
+def initialize_dxgi_output_duplication(
+    dxgi_output: IDXGIOutput1, d3d_device: _Pointer[ID3D11Device]
+) -> _Pointer[IDXGIOutputDuplication]: ...
+def get_dxgi_output_duplication_frame(
+    dxgi_output_duplication: IDXGIOutputDuplication,
+    d3d_device: ID3D11Device,
+    process_func: Callable[[PFLOAT, int, int, int, _ProcessFuncRegionArg, int], _ProcessFuncReturn] | None = ...,
```

Comment:
Same as my comment above where there was another function with a `process_func` argument -- I think the `process_func` callable is missing a parameter. (Also, maybe define a private type alias in `dll/__init__.pyi`, and import it in both places where there are `process_func` arguments? It's quite a complicated type :)

```suggestion
    process_func: Callable[[PFLOAT, int, int, int, int, _ProcessFuncRegionArg, int], _ProcessFuncReturn] | None = ...,
```

Reply:
I miscounted the arguments in both you are right.
I'll do that instead of duplicating the complicated type!